common/app/common/pages_helpers.cljc

```diff
@@ -12,14 +12,67 @@
    [app.common.data :as d]
    [app.common.uuid :as uuid]))
 
+(defn walk-pages
+  "Go through all pages of a file and apply a function to each one"
+  ;; The function receives two parameters (page-id and page), and
+  ;; returns the updated page.
+  [f data]
+  (update data :pages-index #(d/mapm f %)))
+
+(defn select-objects
+  "Get a list of all objects in a page that satisfy a condition"
+  [f page]
+  (filter f (vals (get page :objects))))
+
+(defn update-object-list
+  "Update multiple objects in a page at once"
+  [page objects-list]
+  (update page :objects
+          #(into % (d/index-by :id objects-list))))
+
+(defn get-root-component
+  "Get the root shape linked to the component for this shape, if any"
+  [id objects]
+  (let [obj (get objects id)]
+    (if-let [component-id (:component-id obj)]
+      id
+      (if-let [parent-id (:parent-id obj)]
+        (get-root-component parent-id obj)
+        nil))))
+
 (defn get-children
   "Retrieve all children ids recursively for a given object"
   [id objects]
-  (let [shapes (get-in objects [id :shapes])]
+  (let [shapes (vec (get-in objects [id :shapes]))]
     (if shapes
       (d/concat shapes (mapcat #(get-children % objects) shapes))
       [])))
 
+(defn get-children-objects
+  "Retrieve all children objects recursively for a given object"
+  [id objects]
+  (map #(get objects %) (get-children id objects)))
+
+(defn get-object-with-children
+  "Retrieve a list with an object and all of its children"
+  [id objects]
+  (map #(get objects %) (concat [id] (get-children id objects))))
+
+(defn walk-children
```

Comment:
Seems unused.

Reply:
True, this is no longer used since some of the refactors. I can destroy it.

Reply:
yeah, if we don't need ti is much better just removing it unless it is very very clear that is wil be needed in a near future.

Reply:
ok